src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingMapListController.java

```diff
@@ -0,0 +1,244 @@
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+
+public class TeamMatchmakingMapListController extends NodeController<Pane> {
+
+  private static final int TILE_SIZE = 125;
+  private static final int PADDING = 20;
+
+  private static final Comparator<MapVersion> mapVersionComparator = Comparator.nullsFirst(Comparator.comparing(MapVersion::size))
+                                                          .thenComparing(mapVersion -> mapVersion.map().displayName(),
+                                                                         String.CASE_INSENSITIVE_ORDER);
+
+  private static final Comparator<MatchmakerQueueMapPool> mapPoolComparator = Comparator.comparing(MatchmakerQueueMapPool::minRating, Comparator.nullsFirst(Double::compare))
+                                                                   .thenComparing(MatchmakerQueueMapPool::maxRating, Comparator.nullsLast(Double::compare));
+
+
+  private final MapService mapService;
+  private final UiService uiService;
+  private final PlayerService playerService;
+  private final FxApplicationThreadExecutor fxApplicationThreadExecutor;
+  private final DoubleProperty maxWidth = new SimpleDoubleProperty(0);
+  private final DoubleProperty maxHeight = new SimpleDoubleProperty(0);
+  private final ObjectProperty<SortedMap<MatchmakerQueueMapPool, List<MapVersion>>> sortedBracketsWithDuplicates = new SimpleObjectProperty<>();
+  private final ObjectProperty<SortedMap<MatchmakerQueueMapPool, List<MapVersion>>> sortedBrackets = new SimpleObjectProperty<>();
+  private final ObjectProperty<Integer> playerBracketIndex = new SimpleObjectProperty<>(null);
+  private final ObjectProperty<MatchmakerQueueInfo> queue = new SimpleObjectProperty<>();
+
+  public Pane root;
+  public FlowPane tilesContainer;
+  public ScrollPane scrollContainer;
+  public VBox loadingPane;
+
+  @Override
+  protected void onInitialize() {
+    this.bindProperties();
+  }
+
+  private void bindProperties() {
+    JavaFxUtil.bindManagedToVisible(loadingPane);
+
+    this.queue.addListener((obs, oldVal, newVal) -> {
+      loadingPane.setVisible(true);
+
+      mapService.getMatchmakerBrackets(newVal).subscribe(rawBrackets -> {
+        loadingPane.setVisible(false);
+        this.sortedBracketsWithDuplicates.set(getSortedBrackets(rawBrackets));
+        this.sortedBrackets.set(removeDuplicates(this.sortedBracketsWithDuplicates.get()));
+      });
+    });
```

Comment:
Can be simplified with some of the new javafx apis. This would 

```suggestion
    this.queue.subscribe(value -> {
      loadingPane.setVisible(true);

      mapService.getMatchmakerBrackets(value).subscribe(rawBrackets -> {
        loadingPane.setVisible(false);
        this.sortedBracketsWithDuplicates.set(getSortedBrackets(rawBrackets));
      });
    });
```